docs/adr/0002-shopping-lists-lineitem-and-textlineitem-update-actions.md

```diff
@@ -0,0 +1,687 @@
+  ]
+}
+</pre>
+        </td>
+        <td><pre lang="json">
+{
+  "lineItems": [
+    {
+      "id": "24de3821-e27d-4ddb-bd0b-ecc99365285f",
+      "variant": {
+        "sku": "SKU-1"
+      },
+      "quantity": 1,
+      "custom": {
+        "type": {
+          "id": "4796e155-f5a4-403a-ae1a-04b10c9dfc54",
+          "obj": {
+            "key": "custom-type-for-shoppinglists"
+          }
+        },
+        "fields": {
+          "textField": "text-1"
+        }
+      },
+     "addedAt": "2020-11-04T09:38:35.571Z"
+    },
+    {
+      "id": "24de3821-e27d-4ddb-bd0b-ecc99365285f",
+      "variant": {
+        "sku": "SKU-2"
+      },
+      "quantity": 2,
+      "custom": {
+        "type": {
+          "id": "4796e155-f5a4-403a-ae1a-04b10c9dfc54",
+          "obj": {
+            "key": "custom-type-for-shoppinglists"
+          }
+        },
+        "fields": {
+          "textField": "text-2"
+        }
+      },
+     "addedAt": "2020-11-04T09:40:12.341Z"
+    }
+  ]
+}
+</pre>
+        </td>
+    </tr>
+    <tr>
+        <th colspan="2">Analysis</th>
+    </tr>
+    <tr>
+        <td colspan="2">
+            <p>The Draft has line items with <b>SKU-2</b> and <b>SKU-1</b> also in target project line item
+                with <b>SKU-2</b> and <b>SKU-1</b> exists but in a different order.</p>
+            <p> So we need
+                a <a href="https://docs.commercetools.com/api/projects/shoppingLists#change-lineitems-order">Change
+                    LineItems Order</a>
+                of the line items with order <b>SKU-2</b> and <b>SKU-1</b>.</p>
+            <p>The <b>challenge</b> here is about the line item order and no new line item is added or removed,
+                just order is different, so we need to find a better way to avoid creating unnecessary actions like
+                removing and adding back, is this possible ?
+            </p>
+        </td>
+    </tr>
+    <tr>
+        <th colspan="2">Proposed solution</th>
+    </tr>
+    <tr>
+        <td colspan="2">
+            <p>
+                The solution idea for the changing order with removing and adding back looks like an overhead because
+                we know the all line item ids, so change order action could be created, in this case, the challenge is
+                finding an algorithm to compare and find the line item ids, and then prepare an order.                
+            </p>
+            <p>
+                First example seems reasonable but how you would sync a case like: 
+                '[SKU-1, SKU-2, SKU-3]' to '[SKU-3, SKU-1, SKU-4, SKU-2]', so with a new algorithm it might be done with 
+                change order [line-item-id-3, line-item-id-1, line-item-id-2] then removeLineItem SKU-2, add back addLineItem SKU-4, 
+                in total 3 actions.
+            </p>
+            <p>
+                It looks like there are more different cases and that's why we decided to keep the idea of removing and adding back to 
```

Comment:
```suggestion
                It looks like there are more different cases. That's why we decided to keep the idea of removing and adding back to 
```

Reply:
Make the conclusion more dominant (bold, bigger font).

Reply:
now better: https://github.com/commercetools/commercetools-sync-java/pull/614/commits/98822a91973461658f69fc5e990d58afad211739